src/ControlSystem/ControlErrors/SizeControlStateAhSpeed.cpp

```diff
@@ -0,0 +1,112 @@
+
+  // Note that delta_radius_is_in_danger and char_speed_is_in_danger
+  // can be different for different SizeControlStates.
+  //
+  // The value of 20 causes the control system to panic easily if
+  // delta_radius is decreasing quickly.
+  constexpr double time_tolerance_for_delta_r_in_danger = 20.0;
+  const bool delta_radius_is_in_danger =
+      crossing_time_info.horizon_will_hit_excision_boundary_first and
+      crossing_time_info.t_delta_radius <
+          info->damping_time * time_tolerance_for_delta_r_in_danger;
+
+  const bool char_speed_is_in_danger = [&crossing_time_info, &info,
+                                        &delta_radius_is_in_danger,
+                                        &min_char_speed]() {
+    // speed_tolerance_for_char_speed_in_danger is slightly greater than
+    // unity so that we don't panic if the actual char speed is large
+    // enough compared to the target char speed.
+    constexpr double speed_tolerance_for_char_speed_in_danger = 1.1;
+    // We don't want to panic unless crossing time is less than
+    // the current damping time.
+    constexpr double time_tolerance_for_char_speed_in_danger = 0.99;
+    if (delta_radius_is_in_danger) {
+      return false;
+    } else if (crossing_time_info.char_speed_will_hit_zero_first and
+               crossing_time_info.t_char_speed <
+                   info->damping_time *
+                       time_tolerance_for_char_speed_in_danger and
+               min_char_speed < info->target_char_speed *
+                                    speed_tolerance_for_char_speed_in_danger) {
+      return true;
+    } else {
+      return false;
+    }
+  }();
+
+  const bool comoving_decreasing_slower_than_char_speeds =
+      not(crossing_time_info.t_char_speed > 0.0 &&
+          crossing_time_info.t_comoving_char_speed > 0.0 &&
+          update_args.min_comoving_char_speed > 0.0 &&
+          update_args.min_comoving_char_speed /
+                  crossing_time_info.t_comoving_char_speed >
+              update_args.min_char_speed / crossing_time_info.t_char_speed);
+
+  if (char_speed_is_in_danger) {
+    constexpr double min_char_speed_increase_factor = 1.01;
+    if (info->target_char_speed < min_char_speed) {
+      // We are already in state 1, and we are in danger.
+      // But TargetSpeed is less than CharSpeedMin, so we don't want
+      // to continue to push CharSpeedMin downward.  Instead, increase
+      // TargetSpeed to be above CharSpeedMin.
+      // We don't do this if DeltaRexIsInDanger, because for DeltaRex
+      // in danger, we might need to drive CharSpeed lower.
+      info->target_char_speed = min_char_speed * min_char_speed_increase_factor;
+    }
+    info->suggested_time_scale = crossing_time_info.t_char_speed;
+  } else if (delta_radius_is_in_danger) {
+    constexpr double target_speed_decrease_factor = 0.125;
+    constexpr double time_tolerance_for_delta_r =
+        0.25 * time_tolerance_for_delta_r_in_danger;
```

Comment:
Why these values? (Again, experimentation in SpEC is valid) Same for the rest of the `constexpr` values in this PR that aren't self explanatory (like `Y00`)